src/test/java/com/marklogic/spark/writer/StreamRowsTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.spark.writer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.spark.AbstractIntegrationTest;
+import com.marklogic.spark.Options;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.streaming.StreamingQuery;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class StreamRowsTest extends AbstractIntegrationTest {
+    @Test
+    void streamRowsFromCsvFile() throws TimeoutException {
+        SparkSession sparkSession = newSparkSession();
+        StructType struct = new StructType()
+            .add("Name", DataTypes.StringType)
+            .add("House", DataTypes.StringType);
+
+        Dataset<Row> dataset = sparkSession.readStream().schema(struct)
+            .option("header", true)
+            .format("csv")
+            .load("src/test/resources/inputForStream"); // Needs to be a directory when using streams
+
+        StreamingQuery query = dataset.writeStream()
+            .outputMode("append") // TODO: Investigate for complete and update modes
+            .format(Options.FORMAT)
+            .option(Options.HOST, testConfig.getHost())
+            .option(Options.PORT, testConfig.getRestPort())
+            .option(Options.USERNAME, testConfig.getUsername())
+            .option(Options.PASSWORD, testConfig.getPassword())
+            .option(Options.WRITE_COLLECTIONS, "my-test-data")
+            .option(Options.CheckpointLocation,"src/test/resources/checkpoint/"+
```

Comment:
This results in non-gitignored files being added to the project. I think TempDir is a good fit here - just add this as an argument to the method:

    void streamRowsFromCsvFile(@TempDir Path tempDir)

And then you can do this in the option:

    .option(Options.CheckpointLocation, tempDir.toFile().getAbsolutePath())

We don't care about the contents of that directory at all, at least not in the scope of this test. So it's fine for it to be a temp directory, which also gets deleted by default after the test ends.